buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -262,45 +262,47 @@ private void setDistributionType(ElasticsearchDistribution distribution, TestDis
         }
     }
 
-    @Override
-    public void plugin(RegularFileProperty plugin) {
-        this.plugins.add(plugin.map(p -> p.getAsFile().toURI()));
+    // package protected so only TestClustersAware can access
+    @Internal
+    List<Configuration> getPluginAndModuleConfigurations() {
+        return pluginAndModuleConfigurations;
     }
 
-    @Override
-    public void plugin(Provider<URI> plugin) {
-        requireNonNull(plugin, "Plugin name can't be null");
-        checkFrozen();
-        if (plugins.contains(plugin)) {
-            throw new TestClustersException("Plugin already configured for installation " + plugin);
+    // creates a configuration to depend on the given plugin project, then wraps that configuration
+    // to grab the zip as a file provider
+    private Provider<RegularFile> maybeCreatePluginOrModuleDependency(String path) {
+        Project depProject = project.project(path);
+        Configuration configuration = project.getConfigurations().findByName(depProject.getName());
```

Comment:
Should we use something more unique here? Any chance of collisions by just using the project name?

Reply:
Perhaps this is a good use case for using a detached configuration for a few reasons:

1. We don't want folks to fiddle with this configuration in any way
2. We dont' want it accidentally attempted to resolve externally, such as via `resolveAllDependencies` task
3. The only thing that needs a reference to it is this `ElasticsearchNode` itself, so there's no benefit to it being accessible via the projects `DependencyHandler`
4. We avoid the name conflict issue I mention above.

If the concern is the same dependency being added twice we could track the list of project paths added as a separate `Set`.

Reply:
The detached configuration actually makes this a lot cleaner. Done.